### DISCUSSION TITLE
Column visibility and Column reorder with persistance states

### DISCUSSION THREAD

**Discussion by pirex360** [Power-Components/livewire-powergrid#427] [Ideas]
## Feature Request

Make column visibility persists on the browser or within the user, also you can make column reorder the same way: persistence.

Brief description of what is at stake.

### Why is this needed?

Better control of large datasets, with a lot of columns.

**Comment by luanfreitasdev** [comment]
Do you have any idea how it can be done or what technology to use?

**Reply by pirex360** [reply]
Well, I think plain javascript or trying alpine will do the trick. I used to work with datatables.net, a javascript library that has this option, extended as a plugin.
https://datatables.net/plug-ins/index

**Comment by luanfreitasdev** [comment]
Have you tried using persist?

https://livewire-powergrid.com/#/table/features-setup?id=persistarray-items

`<livewire:filters-table :persist="['columns', 'filters']"/>`

**Comment by pirex360** [comment]
Sorry I didn't search correctly the documentation... I just ended up with column order, like swap column order.
I tried to persist and works like a diamond, good implementation ;) great work.

**Reply by pirex360** [reply]
Just one question.... The persist state is user based...or is only for the browser set ? If user A hides column 3, if other user login in the same machine , he will see colum. 3 hidden ? Or is user based ?

**Reply by luanfreitasdev** [reply]
It is not based on user but on table name with columns. The other user won't see the column either unless they toggle the column

**Reply by pirex360** [reply]
So, If I put the id of the table like `id="table".$user_id`, I could make the persists status be user-based...

Is this possible to define the id of the power-grid table ? or the table name is defined by the `wire:id="GZyqDdcHU8wV0R17KOQv"` of the livewire component ?

How we define a dynamic id/tablename  ? I read the documentation and there's no reference to this.
